cli/keyring.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"crypto/rand"
+	"errors"
+	"syscall"
+
+	"github.com/99designs/keyring"
```

Comment:
question: How confident are we in this dependency?  The repo hasn't been updated in more than two years, and it is managed by a company that looks to have a focus way away from the security world and may lack the technical expertise or attention to keep it secure (assuming it and its dependencies were 2 years ago).

99Designs:
> Global creative platform for designers and clients

@jsimnz @nasdf /All

Reply:
I've replaced this dependency with a custom implementation using a newer library.

Reply:
Dependencies and code of the new one look much healthier, thanks Keenan :)